servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -775,6 +777,21 @@ public FileBlockInfo getFileBlockInfo(long fileId, int fileBlockIndex)
   private FileBlockInfo generateFileBlockInfo(InodeFile file, BlockInfo blockInfo)
       throws InvalidPathException {
     // This function should only be called from within synchronized (mInodeTree) blocks.
+
+    // Duplicates BlockInfo with address represented as IP address, for data locality based on IP
+    // address in computation frameworks, like Spark.
+    List<BlockLocation> locations = Lists.newArrayList(blockInfo.locations);
+    for (BlockLocation location : locations) {
+      try {
+        BlockLocation locationWithHostResolved = new BlockLocation(location);
+        NetAddress address = locationWithHostResolved.getWorkerAddress();
+        address.setHost(InetAddress.getByName(address.getHost()).getHostAddress());
+        blockInfo.addToLocations(locationWithHostResolved);
+      } catch (UnknownHostException e) {
+        LOG.error("IP not found for block host address " + location.getWorkerAddress().getHost());
+      }
+    }
```

Comment:
Do we know if this will cause any issues (since the block will appear to exist twice)? Also, could you add a unit test for this?